src/mpf.rs

```diff
@@ -119,11 +119,11 @@ impl Mpf {
         }
     }
 
-    pub fn get_str(&mut self, n_digits: i32, base: i32, exp: &mut c_long) -> String{
+    pub fn get_str(&self, n_digits: i32, base: i32, exp: &mut c_long) -> String {
         let c_str = CString::new("").unwrap();
         let out;
         unsafe{
-            out = CString::from_raw(__gmpf_get_str(c_str.into_raw(), exp, base, n_digits, &mut self.mpf));
+            out = CString::from_raw(__gmpf_get_str(c_str.into_raw(), exp, base, n_digits, &self.mpf));
```

Comment:
Your tests have exposed the (pre-existing) bug that this is a blatant buffer overflow of `c_str`. The `mpf_get_str` documentation says “The area at `str` has to have space for the largest possible number represented by a `s1n` long limb array, plus one extra character.”